data-integration/admin_units.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Oct 22 17:53:54 2020
+
+@author: giuseppeperonato
+"""
+
+import geopandas as gpd
+import pandas as pd
+import sqlalchemy as sqla
+import utilities
+import logging
+
+# GISCO datasets GEOJSON EPSG:4326 1:1milion
+datasets = {
+    "countries": "https://gisco-services.ec.europa.eu/distribution/v2/countries/geojson/CNTR_RG_01M_2020_{}.geojson",
+    "nuts": "https://gisco-services.ec.europa.eu/distribution/v2/nuts/geojson/NUTS_RG_01M_2021_{}.geojson",
+    "lau": "https://gisco-services.ec.europa.eu/distribution/v2/lau/geojson/LAU_RG_01M_2019_{}.geojson",
+}
+
+# Dataset id
+ds_id = 0
+
+
+def get(datasets: dict = datasets, crs: str = "EPSG:4326") -> gpd.GeoDataFrame:
+    """
+    Retrieve NUTS, LAU and countries from GISCO API and make a single, consistent GDF.
+
+    Parameters
+    ----------
+    datasets : dict
+        dict with API URLs.
+    crs : str, optional
+        Spatial Reference System. The default is "EPSG:4326".
+
+    Returns
+    -------
+    admin_units : GeoDataFrame
+        Table with all administrative units.
+
+    """
+    source_crs_code = crs.split(":")[-1]
+    logging.info("Downloading countries...")
+    countries = gpd.read_file(datasets["countries"].format(source_crs_code), crs=crs)
+    logging.info("Downloading NUTS...")
+    nuts = gpd.read_file(datasets["nuts"].format(source_crs_code), crs=crs)
+    logging.info("Downloading LAU...")
+    lau = gpd.read_file(datasets["lau"].format(source_crs_code), crs=crs)
+    logging.info("Done.")
+
+    # Create consistent columns across ds
+    lau = lau.rename({"LAU_NAME": "NAME"}, axis=1)
+    lau["LEVL_CODE"] = 4
+
+    nuts = nuts.rename({"NAME_LATN": "NAME"}, axis=1)
+
+    countries = countries.rename({"CNTR_NAME": "NAME"}, axis=1)
+    countries = countries.rename({"CNTR_ID": "CNTR_CODE"}, axis=1)
+    countries["LEVL_CODE"] = 0
```

Comment:
something that can be done here is split this function into three separate call:

0) nuts download and formatting
0) lau download and formatting
0) country download and formatting

Reply:
Yes, but it think it is easier like that, they populate the same table anyway